source/adapters/hip/device.cpp

```diff
@@ -950,8 +951,57 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetNativeHandle(
 }
 
 UR_APIEXPORT ur_result_t UR_APICALL urDeviceCreateWithNativeHandle(
-    ur_native_handle_t, ur_platform_handle_t,
-    const ur_device_native_properties_t *, ur_device_handle_t *) {
+    ur_native_handle_t hNativeDevice, ur_platform_handle_t hPlatform,
+    const ur_device_native_properties_t *pProperties,
+    ur_device_handle_t *phDevice) {
+  std::ignore = pProperties;
```

Comment:
I know that's probably how it is in the CUDA adapter, but we should use `[[maybe_unused]]` instead of `std::ignore` for new code, and eventually we should probably fix the old code to use that too.

Reply:
Aye, good idea. Should I fix up the same situation in the CUDA adapter while I'm "here", or does that cause more trouble than it's worth?

Reply:
I wouldn't fix it for the CUDA adapter in this patch, it'd probably be better as a separate NFC patch fixing all of them at once